### PR TITLE
feat: fetch leaderboard from backend

### PR DESCRIPTION
**Pull request by schmanu**
## Content
- Adds leaderboard and rank endpoints
- fixes pagination for lock history
- small design tweaks in the leaderboard

### CODE REVIEW

**Review by usame-algan — APPROVED** [review]
Should we add a loading indicator for the leaderboard? There is some overlap of elements at the moment.

Screenshot 2024-03-25 at 15 50 51

File: src/components/TokenLocking/Leaderboard.tsx [hunk 1]
```diff
@@ -46,71 +50,121 @@ const StyledTableRow = styled(TableRow)(({ theme }) => ({
...
+
+  if (ownRank) {
+    return (
+      <HighlightedTableRow key={ownRank.holder}>
+        <StyledTableCell align="center">
+          {ownRank.position <= 3 ? (
+            <SvgIcon
+              component={
+                ownRank.position === 1 ? FirstPlaceIcon : ownRank.position === 2 ? SecondPlaceIcon : ThirdPlaceIcon
+              }
```
Comment (usame-algan) [inline src/components/TokenLocking/Leaderboard.tsx:106]: nit: We have this logic in multiple places. Could also be an enum or an object to access the correct icon based on the position.

File: src/components/TokenLocking/Leaderboard.tsx [hunk 2]
```diff
@@ -46,71 +50,121 @@ const StyledTableRow = styled(TableRow)(({ theme }) => ({
...
-      amountLocked: 600000,
-    },
-    {
-      rank: 6,
-      address: '0x8803523c9Cc0F948C66FA906ac9065CB03fDc9A3',
-      name: undefined,
-      amountLocked: 505238,
-    },
-  ]
+  const [pages, setPages] = useState([0])
```
[the remark this answers is not in the document]
Comment (schmanu) [reply src/components/TokenLocking/Leaderboard.tsx:167]: Yes exactly. But thinking about it it looks a bit strange :D we could also refactor it to be just a number `pageCount` and create an array out of it.

**Review by usame-algan — APPROVED** [review]
🚀

**Pull request merged**